Handle tfm files which do not contain coding information

juliangilbey opened the pull request:
Hi Han-Wen!

It turns out that converting to Python 3 revealed an ancient issue: the code does not handle TFM files which do not have coding information in a particularly meaningful way, and that causes it to bomb out under Python 3 (see [Debian bug#941885](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=941885)).  In the Python 2 version, the coding string read from the file was meaningless binary, which didn't match any recognised coding system, but in Python 3, the encoding of the string needs to be specified, and so it breaks.

This patch handles this by only reading the coding string if the header is more than 2 words long.

Best wishes,

Julian

hanwen commented:
can you provide more detail? Which tfm files lack coding? 

Is there a way you can do a test run from the make file on one of these files so this new code gets exercised?

juliangilbey commented:
Hi Han-Wen,

If you run (as the Debian bug reporter did) "mftrace lhr10", the current version bombs out, but it works with the patched version.

Best wishes,

   Julian

juliangilbey commented:
Adding cmr11 to the TEST_FONTS line in the makefile does the job; added this to the patch.
Also, in my experiments, I discovered that mftrace crashes if you ask it to convert more than one font at once, because it changes directory out of the temp_dir at the end of do_file, but doesn't change back again on the next do_file run; I've fixed that too.

Best wishes,

Julian
